libr/bin/p/bin_symbols.c

```diff
@@ -502,71 +271,30 @@ static bool load_buffer(RBinFile *bf, void **bin_obj, RBuffer *buf, ut64 loadadd
 		return false;
 	}
 	printSymbolsHeader (sh);
-
-	// 0x40 - contain list of segments
 	SymbolsMetadata sm = parseMetadata (buf, 0x40);
-
-	// 0x138 - 0x220        // unknown information + duplicated list of segments
-	SymbolsDragons sd = parseDragons (buf, sm.addr + sm.size, sm.bits);
-	// eprintf ("sections: %d\n", sd.n_sections);
-	// 0x220 - 0x3a0        // table of sections
-
-	// 0x3a0 - 0x1648       // table of dwords with -1
-	// XXX this is hacky, do not hardcode
-	symbolsBeginAt = sd.addr + sd.size; // is64? 0x458: 0x3a0;
-	D eprintf ("Symbols at 0x%08x\n", (ut32)symbolsBeginAt);
-	RList *symbols = parseSymbols (buf, symbolsBeginAt, &dwordsBeginAt, sd.n_symbols);
-	D eprintf ("Dwords at 0x%08x\n", (ut32)dwordsBeginAt);
-	stringsBeginAt = dwordsBeginAt + (sd.n_symbols * 8);
-	D eprintf ("Strings at 0x%08x\n", (ut32)stringsBeginAt);
-
-	// 0x1648 - 0x1c80      // table of dword pairs (unknown data)
-	parseTable3 (buf, dwordsBeginAt);
-
-	// 0x1c80 - EOF         // strings
-	RList *strings = parseStrings (buf, stringsBeginAt, stringsBeginAt + r_buf_size (buf));
-	// RList *secs = parseSections (buf, SECTIONS_BEGIN, sd.n_sections, strings);
-	// r_list_free (secs);
-	if (strings) {
-		RList *symbolStrings = filterSymbolStrings (strings, sd.n_sections);
-		//	eprintf ("Count strings: %d\n", r_list_length (strings));
-		//	eprintf ("Symbol strings: %d\n", r_list_length (symbolStrings));
-		// name the symbols
-		RListIter *iter;
-		RBinSymbol *sym;
-		int n = 0; // sections count
-		r_list_foreach (symbols, iter, sym) {
-			int m = n + sd.n_sections;
-			RBinString *bs = r_list_get_n (symbolStrings, m);
-			if (bs) {
-				sym->name = strdup (bs->string);
-			} else {
-				sym->name = r_str_newf ("__unnamed_%d", n);
-			}
-			sym->ordinal = n;
-			n++;
-		}
-		r_list_free (strings);
-		r_list_free (symbolStrings);
-		globalSymbols = symbols;
+	RCoreSymCacheElement *element = parseDragons (bf, buf, sm.addr + sm.size, sm.bits);
+	if (!element) {
```

Comment:
Imho its better to be positive

If (element){ binobj...; return true}
Return false